packages/bolt-interactions/src/interaction-flow.spec.ts

```diff
@@ -0,0 +1,159 @@
+import JestReceiver from '@slack-wrench/jest-bolt-receiver';
+import { App, ConversationStore, MemoryStore } from '@slack/bolt';
+import delay from 'delay';
+
+import { InteractionFlow, interactionFlow } from './index';
+
+const random = 'RANDOM';
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+const defaultInteractionIdGenerator = InteractionFlow.interactionIdGenerator;
+
+describe('Bolt interaction flows', () => {
+  const flowName = 'test';
+  const flowId = InteractionFlow.createFlowId(flowName, random);
+
+  let store: ConversationStore;
+  let receiver: JestReceiver;
+  let app: App;
+  let state: any;
+
+  beforeEach(() => {
+    InteractionFlow.interactionIdGenerator = () => random;
+
+    store = new MemoryStore();
+    InteractionFlow.store = store;
+    receiver = new JestReceiver();
+    state = { foo: 'bar' };
+
+    app = new App({ receiver, token: 'TOKEN' });
+  });
+
+  afterEach(() => {
+    InteractionFlow.interactionIdGenerator = defaultInteractionIdGenerator;
+  });
+
+  it('prevents duplicate flow names', () => {
+    expect.assertions(1);
+
+    interactionFlow(flowName, noop)(app);
+
+    expect(() => interactionFlow(flowName, noop)(app)).toThrow(
+      /declared twice/,
+    );
+  });
+
+  it('Allows state to be set from commands', async () => {
+    expect.assertions(2);
+
+    const command = '/command';
+
+    interactionFlow(flowName, flow => {
+      app.command(command, async ({ ack }) => {
+        ack();
+        await flow.start(state);
+      });
+    })(app);
+
+    const { ack } = receiver.sendSlashCommand(command);
+
+    await delay(0);
+
+    expect(ack).toBeCalled();
+    expect(await store.get(flowId)).toEqual(state);
+  });
+
+  it('uses a uuid as a default id generator', () => {
+    expect.assertions(1);
+
+    expect(defaultInteractionIdGenerator()).toMatch(
+      /[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}/,
+    );
+  });
+
+  describe('stateful actions', () => {
+    const buttonId = 'button-id';
+    const action_id = InteractionFlow.createInteractionId(flowId, buttonId);
+
+    beforeEach(async () => {
+      await store.set(flowId, state);
+    });
+
+    it('sets the previous state for stateful actions', async () => {
+      expect.assertions(2);
+
+      interactionFlow(flowName, flow => {
+        flow.action(buttonId, ({ context, ack }) => {
+          expect(context.state).toEqual(state);
+          ack();
+        });
+      })(app);
+
+      const { ack } = receiver.sendBlockButtonAction({
+        action_id,
+      });
```

Comment:
I don't think about format in my code anymore. Likely I had other params in there and deleted them, then prettier didn't adjust. If prettier doesn't fix it I don't fix it.

Honest question, why did you leave a format comment?

Reply:
Care amount was low, just noticed it while reviewing. Thought it was strange-looking, and seemed like a thing prettier would have autofixed. Prettier says both are right, so 🤷‍♂

Format matters because readability, so I leave comments when I think things look strange.

Felt like a strange formatting decision by prettier if you were autoformatting, only reason I left the comment. The whole point of using something like prettier is to think less about this kind of thing and still have readable code, but I still think about it _some_.

More a "feels weird this didn't format" comment. Don't care if you fix at all, so in hindsight not sure it merited a comment, good to know you didn't think it looked strange though.